sdk/provisioning/Generator/src/Model/Resource.cs

```diff
@@ -127,12 +146,36 @@ public override void Generate()
                         writer.Write($" ?? \"{DefaultResourceVersion}\"");
                     }
                     writer.WriteLine(")");
+                    using (writer.Scope("{", "}")) { }
```

Comment:
is this intentionally empty?

Reply:
Yes, this is generating an empty .ctor for us.  In the past we were initializing all the nested members immediately and relying on them being wrapped in `BicepValue<T>` to avoid loading THE ENTIRE OBJECT GRAPH.  We've had to shuffle things around to prevent that and now there's an `Initialize` method that we call before any property is touched.  Let me add an example resource and example model to the PR to help show the generated code diffs.